networkx/algorithms/coloring/greedy_coloring.py

```diff
@@ -46,29 +48,66 @@ def strategy_random_sequential(G, colors):
     return nodes
 
 
-# TODO There is a faster implementation of this algorithm: keep track of
-# the degrees and update them manually each time a node is removed.
 def strategy_smallest_last(G, colors):
     """Returns a list of the nodes of ``G``, "smallest" last.
 
-    Specifically, the node of minimum degree is repeatedly popped from
-    the graph, then this order is reversed.
+    Specifically, the degrees of each node are tracked. From this,
+    the node of minimum degree is repeatedly popped from the graph,
+    updating its neighbors' degrees, then this order is reversed. 
 
     ``G`` is a NetworkX graph. ``colors`` is ignored.
 
-    This implementation of the strategy runs in `O(n^2)` time.
+    This implementation of the strategy runs in `O(|V| + |E|)` time.
```

Comment:
Documentation fixed in 976ae17. There were three strategies for identifying the min node, and `min()` was the least awful. Other ways would be using `itertools.count()` until you found something (which performs horrendously on large, dense graphs) or manually tracking the smallest, which would eventually break down and you'd have to fall back on one of the previous two methods. Given the tradeoff, I prioritized large, dense graphs over those with a huge variation in degree.